yt/sample_data/api.py

```diff
@@ -0,0 +1,174 @@
+"""
+import json
+import re
+from functools import lru_cache
+from itertools import chain
+from pathlib import Path
+
+import pkg_resources
+import requests
+
+from yt.funcs import mylog
+from yt.utilities.on_demand_imports import NotAModule
+
+try:
+    import pooch
+except ImportError:
+    pooch = NotAModule("pooch")
+try:
+    import pandas as pd
+except ImportError:
+    pd = NotAModule("pandas")
+
+num_exp = re.compile(r"[0-9]+")
+byte_unit_exp = re.compile(r"[KMG]?B")
+
+
+def _parse_byte_size(s: str) -> int:
+    """
+    Convert a string size specification to integer byte size.
+
+    This function should be insensitive to case and whitespace.
+    Examples
+    --------
+    >>> parse_byte_size("11B")
+    11
+    >>> parse_byte_size("10 kB")
+    10000
+    >>> parse_byte_size("67.4GB")
+    67400000000
+    """
+    try:
+        s = s.upper()
+    except AttributeError:
+        # input is not a string (likely a np.nan)
+        return pd.NA
+    val = float(re.search(num_exp, s).group())
+    unit = re.search(byte_unit_exp, s).group()
+    exponent = {"B": 0, "KB": 3, "MB": 6, "GB": 9}[unit]
+    return int(val * 10 ** exponent)
+
+
+def _get_archive_filename(url):
+    return url.split("/")[-1]
+
+
+sample_data_registry = json.load(
+    pkg_resources.resource_stream("yt", "sample_data_registry.json")
+)
+
+
+@lru_cache
+def get_data_registry_table():
+    """
+    Load the sample data registry as a pandas.Dataframe instance.
+
+    This function is considered experimental and is exposed for exploratory purposed.
+    The output format is subject to change.
+
+    The output of this function is cached so it will only generate one request per session.
+    """
+
+    # it would be nicer to have an actual api on the yt website server,
+    # but this will do for now
+    api_url = "https://raw.githubusercontent.com/yt-project/website/master/data/datafiles.json"  # noqa
+
+    response = requests.get(api_url)
+
+    if not response.ok:
+        raise RuntimeError(
+            "Could not retrieve registry data. Please check your network setup."
+        )
+
+    website_json = json.loads(response.text)
+    # this dict follows this schema: {frontend_name: {flat dataframe-like}}
+
+    columns = ["code", "filename", "size", "url"]
+    website_table = pd.concat(pd.DataFrame(d) for d in website_json.values())[columns]
+
+    # add a int-type byte size column
+    website_table["byte size"] = website_table["size"].apply(_parse_byte_size)
+
+    # normalize urls to match the local json
+    website_table["url"] = website_table["url"].apply(
+        lambda u: u.replace("http", "https")
+    )
+
+    # load local data
+    pooch_json = json.load(
+        pkg_resources.resource_stream("yt", "sample_data_registry.json")
+    )
+    pooch_table = pd.DataFrame(pooch_json.values())
+
+    # merge tables
+    unified_table = website_table.merge(pooch_table, on="url", how="outer")
+
+    # unified_table.set_index("filename", inplace=True)
```

Comment:
maybe add a comment here explaining why you're leaving the commented code (I believe it's the same reason as the note in `load_sample`)?

Reply:
exactly ! thank you so much for figuring this out. I'll update this.